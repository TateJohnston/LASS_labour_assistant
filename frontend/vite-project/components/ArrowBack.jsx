import ArrowCircleLeftIcon from "@mui/icons-material/ArrowCircleLeft";
import { Colors } from "../src/assets/Colors";

const ArrowBack = ({
  fontSize = "30px",
  onClick,
  height = "45px",
  width = "45px",
  backgroundColor,
}) => {
  return (
    <ArrowCircleLeftIcon
      onClick={onClick}
      sx={{
        height,
        width,
        color: Colors.secondary,
        backgroundColor,
        fontSize: { fontSize },
        "&:hover": {
          cursor: "pointer",
          transition: "transform ease 0.3s",
          transform: "scale(1.2)",
        },
      }}
    />
  );
};

export default ArrowBack;
