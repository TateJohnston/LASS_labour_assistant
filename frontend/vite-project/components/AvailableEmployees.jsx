import { Box, Typography } from "@mui/material";
import { Colors } from "../src/assets/Colors";

const AvailableEmployee = ({ employee }) => {
  const shiftColors = {
    D: Colors.warning,
    E: Colors.secondary,
    N: Colors.error,
  };

  return (
    <Box
      key={employee.employee_id}
      sx={{
        display: "flex",
        flexDirection: "row",
        justifyContent: "space-between",
        alignItems: "center",
        padding: "15px 0px",
      }}
    >
      <Box
        sx={{
          flex: "0.2",
          display: "flex",
          flexDirection: "row",
        }}
      >
        <Typography
          sx={{ color: shiftColors[employee.shift[0]], fontWeight: "bold" }}
          variant="h5"
        >
          {employee.shift[0]}
        </Typography>
      </Box>
      <Box
        sx={{
          flex: "1",
          display: "flex",
          flexDirection: "row",
          alignItems: "start",
        }}
      >
        <Typography variant="h6">{employee.employee_name}</Typography>
      </Box>
      <Box
        sx={{
          flex: "1.2",
          display: "flex",
          flexDirection: "row",
          gap: "10px",
        }}
      >
        {[
          employee.foreman && "Foreman",
          employee.crane && "Crane",
          employee.clerk && "Clerk",
          employee.fork && "Fork",
          employee.truck && "Truck",
        ]
          .filter(Boolean)
          .map((skill, index) => (
            <Typography key={index}>{skill}</Typography>
          ))}
      </Box>
    </Box>
  );
};
export default AvailableEmployee;
