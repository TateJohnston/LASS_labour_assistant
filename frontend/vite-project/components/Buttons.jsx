import { Button } from "@mui/material";
import { Colors } from "../src/assets/Colors";

const Buttons = ({
  onClick,
  variant,
  width = "250px",
  padding,
  content,
  height = "50px",
  backgroundColor = Colors.secondary,
  fontSize,
}) => {
  return (
    <Button
      onClick={onClick}
      variant={variant}
      sx={{
        fontSize,
        backgroundColor: { backgroundColor },
        color: Colors.content,
        fontWeight: "bold",
        borderRadius: "5px",
        width: { width },
        height: { height },
        padding: { padding },
      }}
    >
      {content}
    </Button>
  );
};

export default Buttons;
