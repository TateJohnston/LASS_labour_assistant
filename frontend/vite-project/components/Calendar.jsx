import FullCalendar from "@fullcalendar/react";
import dayGridPlugin from "@fullcalendar/daygrid";
import interactionPlugin from "@fullcalendar/interaction";

const Calendar = ({
  events,
  eventClick,
  dateClick,
  eventBackgroundColor,
  height,
}) => {
  return (
    <FullCalendar
      height={height}
      plugins={[dayGridPlugin, interactionPlugin]}
      initialView="dayGridMonth"
      events={events}
      eventClick={eventClick}
      eventBackgroundColor={eventBackgroundColor}
      dateClick={dateClick}
    />
  );
};

export default Calendar;
