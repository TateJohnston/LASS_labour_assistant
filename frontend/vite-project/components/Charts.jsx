import { Chart as ChartJS, Legend, Ticks } from "chart.js/auto";
import { Bar, Doughnut, Line } from "react-chartjs-2";
import { Box } from "@mui/material";

export const BarChart = ({
  labels,
  datasets,
  xAxisText,
  yAxisText,
  backgroundColor,
  xAxisTextFontSize,
  yAxisTextFontSize,
  showLegend = true,
  tickFontSize,
  ticksPadding,
}) => {
  return (
    <Box
      sx={{
        width: "80vw",
        height: "100%",
        padding: "15px",
        borderRadius: "5px",
        paddingRight: "40px",
      }}
    >
      <Bar
        data={{
          labels,
          datasets,
          backgroundColor,
        }}
        options={{
          plugins: { legend: { display: showLegend } },
          responsive: true,
          maintainAspectRatio: true,
          scales: {
            x: {
              title: {
                display: true,
                text: xAxisText,
                font: { size: xAxisTextFontSize, weight: "bold" },
                padding: { top: 15 },
              },
              ticks: {
                padding: ticksPadding,
                font: { size: tickFontSize },
              },
            },

            y: {
              title: {
                display: true,
                text: yAxisText,
                font: { size: yAxisTextFontSize, weight: "bold" },
              },
            },
          },
        }}
      />
    </Box>
  );
};

export const DoughnutChart = ({
  labels,
  datasets,
  xAxisText,
  yAxisText,
  backgroundColor,
  xAxisTextFontSize,
  yAxisTextFontSize,
  showLegend = true,
}) => {
  return (
    <Box
      sx={{
        width: "40vw",
        height: "100%",
        borderRadius: "5px",
        paddingRight: "40px",
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
      }}
    >
      <Doughnut
        data={{
          labels,
          datasets,
          backgroundColor,
        }}
      />
    </Box>
  );
};
