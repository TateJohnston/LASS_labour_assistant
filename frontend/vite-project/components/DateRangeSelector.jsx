import React, { useState } from "react";
import { DateRange } from "react-date-range";
import { addDays, isAfter } from "date-fns";
import "react-date-range/dist/styles.css";
import "react-date-range/dist/theme/default.css";
import { Colors } from "../src/assets/Colors";

const DateRangeSelector = ({ onChange, open, toggle }) => {
  const [range, setRange] = useState([
    {
      startDate: undefined,
      endDate: undefined,
      key: "selection",
    },
  ]);

  const handleSelect = (ranges) => {
    const selected = ranges.selection;
    setRange([selected]);
    if (
      selected.startDate &&
      selected.endDate &&
      isAfter(selected.endDate, selected.startDate)
    ) {
      onChange(selected);
    }
  };

  if (!open) return null;

  return (
    <div
      style={{
        display: "flex",
        flexDirection: "column",
        position: "absolute",
        top: "55px",
        left: "50%",
        transform: "translateX(-50%)",
        background: "white",
        padding: "16px",
        border: `1px solid ${Colors.primary}`,
        borderRadius: "8px",
        zIndex: 10,
      }}
    >
      <DateRange
        editableDateInputs
        onChange={handleSelect}
        moveRangeOnFirstSelection={false}
        ranges={range}
      />
      <button onClick={toggle} style={{ marginTop: "10px" }}>
        Close
      </button>
    </div>
  );
};

export default DateRangeSelector;
