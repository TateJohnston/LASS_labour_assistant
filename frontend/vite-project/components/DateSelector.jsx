import * as React from "react";
import { DatePicker } from "@mui/x-date-pickers";
import { LocalizationProvider } from "@mui/x-date-pickers";
import { AdapterDayjs } from "@mui/x-date-pickers/AdapterDayjs";
import { Colors } from "../src/assets/Colors";

const DateSelector = ({
  height,
  onChange,
  label = "Select Date",
  width = "300px",
}) => {
  return (
    <LocalizationProvider dateAdapter={AdapterDayjs}>
      <DatePicker
        sx={{
          height,
          backgroundColor: "white",
          width,
          borderRadius: "5px",
          textAlign: "center",
        }}
        onChange={(newValue) => onChange(newValue)}
        label={label}
      />
    </LocalizationProvider>
  );
};

export default DateSelector;
