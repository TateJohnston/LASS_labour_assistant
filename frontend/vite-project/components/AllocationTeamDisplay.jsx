import { Box, Typography } from "@mui/material";
import { Colors } from "../src/assets/Colors";
import Buttons from "./Buttons";
import { useState } from "react";
import DialogueBox from "./DialogueBox";
import axios from "axios";

const AllocationTeamDisplay = ({ team_id, employees, fetchTeams, date }) => {
  const [openDialogue, setOpendialogue] = useState(false);

  const removeTeam = (team_id) => {
    axios
      .delete(`http://localhost:8081/lass/teams/deleteTeam/${team_id}`)
      .then((data) => {
        fetchTeams();
      })
      .catch((err) => {
        console.log(err);
      });
  };

  return (
    <>
      <Box
        key={team_id}
        sx={{
          width: "80%",
          display: "flex",
          flexDirection: "column",
          backgroundColor: Colors.primary,
          color: Colors.secondary,
          borderRadius: "20px",
          padding: "20px",
        }}
      >
        <div
          style={{
            height: "20px",
            width: "20px",
            backgroundColor: Colors.error,
            fontWeight: "bold",
            display: "flex",
            justifyContent: "center",
            alignItems: "center",
            color: "black",
            borderRadius: "5px",
            cursor: "pointer",
          }}
          onClick={() => setOpendialogue(true)}
        >
          X
        </div>
        <DialogueBox
          open={openDialogue}
          onClose={() => setOpendialogue(false)}
          confirmClick={() => {
            removeTeam(team_id);
            setOpendialogue(false);
          }}
          dialogueTitle={`Remove Team ${team_id}?`}
          cancelText={"Cancel"}
          submitText={"Confirm"}
        />
        <Box
          sx={{
            display: "flex",
            flexDirection: "row",
            justifyContent: "center",
            gap: "5px",
            alignItems: "center",
            color: Colors.content,
            marginBottom: "5px",
          }}
        >
          <Typography sx={{}} variant="h5">
            {employees[0].shift}:
          </Typography>
          <Typography sx={{}} variant="h5">
            Team {team_id}
          </Typography>
        </Box>

        {employees[0].name === null ? (
          <Typography
            sx={{
              flex: "1",
              padding: "5px",
              textAlign: "center",
              color: "darkgray",
              fontStyle: "italic",
            }}
          >
            No employees added yet...
          </Typography>
        ) : (
          employees.map((employee) => (
            <Box
              key={employee.employee_id}
              sx={{
                display: "flex",
                flexDirection: "row",
                alignItems: "center",
                justifyContent: "center",
              }}
            >
              <Typography
                sx={{
                  flex: "1",
                  padding: "5px",
                  textAlign: "end",
                  fontWeight: "bold",
                  color: "lightgray",
                }}
              >
                {employee.role}:
              </Typography>
              <Typography
                sx={{
                  flex: "1",
                  padding: "5px",
                  textAlign: "start",
                  fontWeight: "bold",
                  color: Colors.secondary,
                }}
              >
                {employee.name}
              </Typography>
            </Box>
          ))
        )}
      </Box>
    </>
  );
};

export default AllocationTeamDisplay;
