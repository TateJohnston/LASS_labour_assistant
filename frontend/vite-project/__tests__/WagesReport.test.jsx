import { queryByText, render } from "@testing-library/react";
import { it, expect } from "vitest";
import { BarChart } from "../components/Charts";
import { Colors } from "chart.js";

it("Barchart functions correctly", () => {
  const mockData = [
    {
      employee_id: 3,
      employee_name: "Noah Smith",
      role_id: 1,
      salary_YTD: "9218.12",
      bonus_YTD: "2625.69",
      upgrades_YTD: "0.00",
      gross_YTD: "12730.34",
      tax_YTD: "3331.20",
      net_YTD: "9399.14",
      super_YTD: "1527.64",
    },
    {
      employee_id: 3,
      employee_name: "Noah Smith",
      role_id: 1,
      salary_YTD: "9218.12",
      bonus_YTD: "2625.69",
      upgrades_YTD: "0.00",
      gross_YTD: "12730.34",
      tax_YTD: "3331.20",
      net_YTD: "9399.14",
      super_YTD: "1527.64",
    },
    {
      employee_id: 6,
      employee_name: "Henry White",
      role_id: 2,
      salary_YTD: "8257.64",
      bonus_YTD: "3580.45",
      upgrades_YTD: "0.00",
      gross_YTD: "12119.24",
      tax_YTD: "3222.74",
      net_YTD: "8896.49",
      super_YTD: "1454.31",
    },
  ];
  const roleDictionary = {
    1: "Foreman",
    2: "Crane Driver",
    3: "Clerk",
    4: "Truck Driver",
    5: "Forklift Operator",
    6: "Payroll Manager",
    7: "Labour Manager",
  };

  const roleColors = {
    1: Colors.secondary,
    2: Colors.success,
    3: Colors.warning,
    4: Colors.error,
    5: "gray",
    6: "blue",
    7: "lightgray",
  };

  const { asFragment } = render(
    <BarChart
      showLegend={false}
      labels={mockData.map(
        (employee) =>
          `${employee.employee_name} - ${roleDictionary[employee.role_id]}`
      )}
      datasets={[
        {
          data: mockData.map((employee) => employee.gross_YTD),
          backgroundColor: mockData.map(
            (employee) => roleColors[employee.role_id]
          ),
        },
      ]}
      xAxisText={"Employees"}
      yAxisText={"Employees YTD Gross ($)"}
      xAxisTextFontSize={20}
      yAxisTextFontSize={20}
    />
  );

  expect(asFragment()).toMatchSnapshot();
});
