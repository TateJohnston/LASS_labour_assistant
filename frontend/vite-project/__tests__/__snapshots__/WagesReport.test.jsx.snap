// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Barchart functions correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-6vcxt"
  >
    <canvas
      height="150"
      role="img"
      width="300"
    />
  </div>
</DocumentFragment>
`;
